fvtest/compilertest/tests/SingleBitContainerTest.cpp

```diff
@@ -0,0 +1,332 @@
+	
+	//Rest the first bit
+	container.set();
+	container.resetAll(1);
+	ASSERT_EQ(container.get(), false) << "resetAll - bit not reset to false";
+}
+
+
+//****** Test setAll(int64_t m, int64_t n) ******//
+
+TEST_F(SingleBitContainerTest, setAllTest1) {
+	
+	//Set first bit to true
+	container.setAll(0, 1);
+	ASSERT_EQ(container.get(), true) << "setAll - bit not set";
+}
+
+TEST_F(SingleBitContainerTest, setAllTest2) {
+
+	//m = 1, n = 1, container should not be set	
+	container.setAll(1,1);
+	ASSERT_EQ(container.get(), false);
+}
+
+TEST_F(SingleBitContainerTest, setAllTest3) {
+
+	//m = 0, n = 0, container should not be set
+	container.setAll(0,0);
+	ASSERT_EQ(container.get(), false);
+}
+
+
+//******* Test setAll(int64_t n) *******//
+
+TEST_F(SingleBitContainerTest, setAllTest4) {
+	
+	//setting bit to false
+	container.setAll(0);
+	ASSERT_EQ(container.get(), false);
+}
+
+TEST_F(SingleBitContainerTest, setAllTest5){
+	
+	//setting bit to true
+	container.setAll(1);
+	ASSERT_EQ(container.get(), true);
+}
+
+
+//******* Test intersection of two container ******//
+
+TEST_F(SingleBitContainerTest, intersects) {
+
+	//Intersection for two false container
+	container.resetAll(1);
+	other.resetAll(1);
+	ASSERT_EQ(container.intersects(other), false);
+
+	//Intersection of two true bit container
+	container.setAll(1);
+	other.setAll(1);
+	ASSERT_EQ(container.intersects(other), true);
+
+	//Intersectino of one false one true container
+	container.resetAll(1);
+	other.setAll(1);
+	ASSERT_EQ(container.intersects(other), false);
+}
+
+//****** Test operatorEqual ******//
+
+TEST_F(SingleBitContainerTest, operatorEqual){
+	
+	//Both container set to false
+	container.resetAll(1);
+	other.resetAll(1);
+	ASSERT_EQ(container == other, true);
+
+	//Both container set to true
+	container.setAll(1);
+	other.setAll(1);
+	ASSERT_EQ(container == other, true);
+
+	//Equally set of two container set differently
+	container.setAll(1);
+	other.resetAll(1);
+	ASSERT_EQ(container == other, false);
+
+}
+
+//****** Test operatorNotEqual ******//
+
+TEST_F(SingleBitContainerTest, operatorNotEqualTest1) {
+	
+	//Both container set to false
+	container.resetAll(1);
+	other.resetAll(1);
+	ASSERT_EQ(container != other, false);
+
+	//Both container set to true
+	container.setAll(1);
+	other.resetAll(1);
+	ASSERT_EQ(container != other, true);
+}
+
+//****** Test for |= operator ******//
+
+TEST_F(SingleBitContainerTest, bitOperator) {
+	
+	//Both container set to false
+	container |= other;
+	ASSERT_EQ(container.get(), false);
+
+	//Both container set to true
+	container.setAll(1);
+	other.resetAll(1);
+	container |= other;
+	ASSERT_EQ(container.get(), true);
+
+	//other container set to true
+	container.resetAll(1);
+	other.setAll(1);
+	container |= other;
+	ASSERT_EQ(container.get(), true);
+
+}
+
+//****** Test for &= operator ******//
+
+TEST_F(SingleBitContainerTest, operatorAndEqualTest) {
+	
+	//Both container set to false
```

Comment:
Should `container` and `other` be explicitly set to false?  Or are you relying on default behaviour?  Please call that out in a comment if so.